include/sparrow/arrow_interface/arrow_array_schema_utils.hpp

```diff
@@ -21,11 +21,50 @@
 #include <type_traits>
 #include <vector>
 
+#include "sparrow/c_interface.hpp"
 #include "sparrow/utils/memory.hpp"
 #include "sparrow/utils/mp_utils.hpp"
 
 namespace sparrow
 {
+    /**
+     * Release the children and dictionnary of an ArrowArray or ArrowSchema.
+     * 
+     * @tparam T ArrowArray or ArrowSchema
+     * @param t The ArrowArray or ArrowSchema to release.
+     */
+    template <class T>
+    requires std::same_as<T, ArrowArray> || std::same_as<T, ArrowSchema>
+    void release_common_arrow(T* t)
```

Comment:
`t` must never be null so that should be a reference. If you prefer a pointer (I prefer a reference), document that it should not be null and add an assertion OR allow it to be null and just check and return if so (and document it).

Reply:
done